pkg/knowledge_base2/graph.go

```diff
@@ -190,7 +191,7 @@ func Downstream(dag construct.Graph, kb TemplateKB, rid construct.ResourceId, la
 		return nil, fmt.Errorf("unknown layer %s", layer)
 	}
 	err := graph_addons.WalkDown(dag, rid, f)
-	return result, err
+	return result.ToSlice(), err
```

Comment:
note: this will be non-deterministic, not sure if that's important or not

Reply:
i think thats fine, im not sure why we would need it to be deterministic